Firestore/core/src/firebase/firestore/remote/stream_objc_bridge.h

```diff
@@ -89,8 +89,52 @@ class WatchStreamSerializer {
 };
 
 /**
- * A C++ bridge that invokes methods on an `FSTWatchStreamDelegate`.
+ * A C++ bridge to `FSTSerializerBeta` that allows creating
+ * `GCFSWriteRequest`s and parsing `GCFSWriteResponse`s.
  */
+class WriteStreamSerializer {
+ public:
+  explicit WriteStreamSerializer(FSTSerializerBeta* serializer)
+      : serializer_{serializer} {
+  }
+
+  void UpdateLastStreamToken(GCFSWriteResponse* proto);
+  void SetLastStreamToken(NSData* token) {
+    last_stream_token_ = token;
+  }
+  NSData* GetLastStreamToken() const {
+    return last_stream_token_;
+  }
+
+  GCFSWriteRequest* CreateHandshake() const;
+  GCFSWriteRequest* CreateWriteMutationsRequest(
+      NSArray<FSTMutation*>* mutations) const;
+  GCFSWriteRequest* CreateEmptyMutationsList() {
+    return CreateWriteMutationsRequest(@[]);
+  }
+  static grpc::ByteBuffer ToByteBuffer(GCFSWriteRequest* request);
+
+  /**
+   * If parsing fails, will return nil and write information on the error to
+   * `out_status`. Otherwise, returns the parsed proto and sets `out_status` to
+   * ok.
```

Comment:
I tried that first and got an error message from the compiler which basically stated that ARC-counted objects can't be used with `StatusOr`. Added a comment to that effect.

Reply:
FWIW, it seems conventional in C++ to return these values the other way: Status as the return value and actual value as an out parameter. This avoids needing to return a default value in case of error. In case of Objective-C objects that doesn't seem to much benefit though, since nil always works.

No action required, unless you think Status/StatusOr will remain a part of the C++ interface. Also feel free to punt to another PR.